pulp_rpm/tests/functional/api/test_copy.py

```diff
@@ -265,3 +273,138 @@ def test_all_content_recursive(self):
+
+        repo = self.repo_api.read(repo.pulp_href)
+        whale_package_href = [
+            pkg
+            for pkg in get_content(repo.to_dict())['rpm.package']
+            if pkg['name'] == 'whale'
+        ][0]['pulp_href']
+        package_to_copy = []
+        package_to_copy.append(whale_package_href)
+
+        config = [{
+            'source_repo_version': repo.latest_version_href,
+            'dest_repo': empty_repo.pulp_href,
+            'content': package_to_copy
+        }]
+
+        rpm_copy(self.cfg, config, recursive=True)
+        empty_repo = self.repo_api.read(empty_repo.pulp_href)
+        empty_repo_packages = [
+            pkg['name']
+            for pkg in get_content(empty_repo.to_dict())['rpm.package']
+        ]
+
+        # assert that only 3 packages are copied (original package with its two dependencies)
+        self.assertEqual(len(empty_repo_packages), 3)
+        # assert dependecies package names
+        self.assertIn('stork', empty_repo_packages)
+        self.assertIn('shark', empty_repo_packages)
+
+    def test_sctrict_copy_package_to_existing_repo(self):
+        """Test copy package and its dependecies to empty repository.
+
+        - Create repository and populate it
+        - Create second repository with package fullfiling test package dependency
+        - Use 'copy' to copy 'whale' package with dependencies
+        - assert package and its missing dependencies were copied
+        """
+        # prepare final_repo - copy to repository
+        final_repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, final_repo.pulp_href)
+
+        body = gen_rpm_remote(url=RPM_UNSIGNED_FIXTURE_URL)
+        remote = self.remote_api.create(body)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(final_repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+
+        final_repo = self.repo_api.read(final_repo.pulp_href)
+
+        # prepare repository - copy from repository
+        repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+
+        body = gen_rpm_remote(url=RPM_UNSIGNED_FIXTURE_URL)
+        remote = self.remote_api.create(body)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+
+        repo = self.repo_api.read(repo.pulp_href)
+
+        # remove one dependecny package from final repository
+        data = {
+            "remove_content_units": [
+                pkg.pulp_href
+                for pkg in self.rpm_content_api.list().results
+                if pkg.name == "shark" or pkg.name == "whale"
```

Comment:
```if pkg.name in {"shark", "whale"}``` might be slightly better

Reply:
yet added random factor to choose dependency randomly